#### Milestone 2 #####

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [ ]:
We want to load plots, 

In [38]:
df_plot = pd.read_csv("data/MovieSummaries/plot_summaries.txt", sep='\t', header=None, names=["wikiID", "plot"])
df_meta = pd.read_csv("data/MovieSummaries/movie.metadata.tsv", sep='\t', header=None, names=["wikiID", "freeID", "name", "release_date", "revenue", "runtime", "languages", "countries", "genres"])

In [41]:
df_plot

wikiID                                               plot
0      23890098  Shlykov, a hard-working taxi driver and Lyosha...
1      31186339  The nation of Panem consists of a wealthy Capi...
2      20663735  Poovalli Induchoodan  is sentenced for six yea...
3       2231378  The Lemon Drop Kid , a New York City swindler,...
4        595909  Seventh-day Adventist Church pastor Michael Ch...
...         ...                                                ...
42298  34808485  The story is about Reema , a young Muslim scho...
42299   1096473  In 1928 Hollywood, director Leo Andreyev  look...
42300  35102018  American Luthier focuses on Randy Parsons’ tra...
42301   8628195  Abdur Rehman Khan , a middle-aged dry fruit se...
42302   6040782  1940 - Operation Dynamo has just taken place. ...

[42303 rows x 2 columns]

In [40]:
df_meta.head()

wikiID      freeID  \
0        975900   /m/03vyhn   
1       3196793   /m/08yl5d   
2      28463795  /m/0crgdbh   
3       9363483  /m/0285_cd   
4        261236   /m/01mrr1   
...         ...         ...   
81736  35228177  /m/0j7hxnt   
81737  34980460  /m/0g4pl34   
81738   9971909  /m/02pygw1   
81739    913762   /m/03pcrp   
81740  12476867  /m/02w7zz8   

                                                    name release_date  \
0                                         Ghosts of Mars   2001-08-24   
1      Getting Away with Murder: The JonBenét Ramsey ...   2000-02-16   
2                                            Brun bitter         1988   
3                                       White Of The Eye         1987   
4                                      A Woman in Flames         1983   
...                                                  ...          ...   
81736                           Mermaids: The Body Found   2011-03-19   
81737                                            Knuckle   2011-01-21   
81738                                  Another Nice Mess   1972-09-22   
81739  The Super Dimension Fortress Macross II: Lover...   1992-05-21   
81740                                            Spliced         2002   

          revenue  runtime                           languages  \
0      14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1             NaN     95.0  {"/m/02h40lc": "English Language"}   
2             NaN     83.0  {"/m/05f_3": "Norwegian Language"}   
3             NaN    110.0  {"/m/02h40lc": "English Language"}   
4             NaN    106.0   {"/m/04306rv": "German Language"}   
...           ...      ...                                 ...   
81736         NaN    120.0  {"/m/02h40lc": "English Language"}   
81737         NaN     96.0  {"/m/02h40lc": "English Language"}   
81738         NaN     66.0  {"/m/02h40lc": "English Language"}   
81739         NaN    150.0   {"/m/03_9r": "Japanese Language"}   
81740         NaN     86.0  {"/m/02h40lc": "English Language"}   

                                               countries  \
0              {"/m/09c7w0": "United States of America"}   
1              {"/m/09c7w0": "United States of America"}   
2                                 {"/m/05b4w": "Norway"}   
3                         {"/m/07ssc": "United Kingdom"}   
4                                {"/m/0345h": "Germany"}   
...                                                  ...   
81736          {"/m/09c7w0": "United States of America"}   
81737  {"/m/03rt9": "Ireland", "/m/07ssc": "United Ki...   
81738          {"/m/09c7w0": "United States of America"}   
81739                              {"/m/03_3d": "Japan"}   
81740                            {"/m/0d060g": "Canada"}   

                                                  genres  
0      {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  
1      {"/m/02n4kr": "Mystery", "/m/03bxz7": "Biograp...  
2      {"/m/0lsxr": "Crime Fiction", "/m/07s9rl0": "D...  
3      {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...  
4                                {"/m/07s9rl0": "Drama"}  
...                                                  ...  
81736                            {"/m/07s9rl0": "Drama"}  
81737  {"/m/03bxz7": "Biographical film", "/m/07s9rl0...  
81738       {"/m/06nbt": "Satire", "/m/01z4y": "Comedy"}  
81739  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...  
81740  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  

[81741 rows x 9 columns]

The feature connecting dataframes together is the Wikipedia ID. Also there are more metadatas of movies (81741 movies) than plots (42303 movies). We will only keep the metadatas of the movies we know the plot of.

In [44]:
df_full = df_meta.merge(df_plot, how='inner', on="wikiID")
df_full

wikiID      freeID  \
0        975900   /m/03vyhn   
1       9363483  /m/0285_cd   
2        261236   /m/01mrr1   
3      18998739  /m/04jcqvw   
4       6631279   /m/0gffwj   
...         ...         ...   
42199  23851782  /m/06_vb43   
42200  35228177  /m/0j7hxnt   
42201  34980460  /m/0g4pl34   
42202    913762   /m/03pcrp   
42203  12476867  /m/02w7zz8   

                                                    name release_date  \
0                                         Ghosts of Mars   2001-08-24   
1                                       White Of The Eye         1987   
2                                      A Woman in Flames         1983   
3                              The Sorcerer's Apprentice         2002   
4                                            Little city   1997-04-04   
...                                                  ...          ...   
42199                                    The Ghost Train   1941-05-03   
42200                           Mermaids: The Body Found   2011-03-19   
42201                                            Knuckle   2011-01-21   
42202  The Super Dimension Fortress Macross II: Lover...   1992-05-21   
42203                                            Spliced         2002   

          revenue  runtime                           languages  \
0      14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1             NaN    110.0  {"/m/02h40lc": "English Language"}   
2             NaN    106.0   {"/m/04306rv": "German Language"}   
3             NaN     86.0  {"/m/02h40lc": "English Language"}   
4             NaN     93.0  {"/m/02h40lc": "English Language"}   
...           ...      ...                                 ...   
42199         NaN     82.0  {"/m/02h40lc": "English Language"}   
42200         NaN    120.0  {"/m/02h40lc": "English Language"}   
42201         NaN     96.0  {"/m/02h40lc": "English Language"}   
42202         NaN    150.0   {"/m/03_9r": "Japanese Language"}   
42203         NaN     86.0  {"/m/02h40lc": "English Language"}   

                                               countries  \
0              {"/m/09c7w0": "United States of America"}   
1                         {"/m/07ssc": "United Kingdom"}   
2                                {"/m/0345h": "Germany"}   
3                           {"/m/0hzlz": "South Africa"}   
4              {"/m/09c7w0": "United States of America"}   
...                                                  ...   
42199                     {"/m/07ssc": "United Kingdom"}   
42200          {"/m/09c7w0": "United States of America"}   
42201  {"/m/03rt9": "Ireland", "/m/07ssc": "United Ki...   
42202                              {"/m/03_3d": "Japan"}   
42203                            {"/m/0d060g": "Canada"}   

                                                  genres  \
0      {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
1      {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...   
2                                {"/m/07s9rl0": "Drama"}   
3      {"/m/0hqxf": "Family Film", "/m/01hmnh": "Fant...   
4      {"/m/06cvj": "Romantic comedy", "/m/0hj3n0w": ...   
...                                                  ...   
42199  {"/m/0lsxr": "Crime Fiction", "/m/01jfsb": "Th...   
42200                            {"/m/07s9rl0": "Drama"}   
42201  {"/m/03bxz7": "Biographical film", "/m/07s9rl0...   
42202  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...   
42203  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   

                                                    plot  
0      Set in the second half of the 22nd century, th...  
1      A series of murders of rich young women throug...  
2      Eva, an upper class housewife, becomes frustra...  
3      Every hundred years, the evil Morgana  returns...  
4      Adam, a San Francisco-based artist who works a...  
...                                                  ...  
42199  {{plot}} The film opens with a Great Western e...  
42200  Two former National Oceanic Atmospheric Admin